src/OpenTelemetry.Instrumentation.AspNet/AspNetInstrumentationOptions.cs

```diff
@@ -35,8 +35,13 @@ public class AspNetInstrumentationOptions
         });
 
         /// <summary>
-        /// Gets or sets a hook to exclude calls based on domain or other per-request criterion.
+        /// Gets or sets a Filter function to filter instrumentation for requests on a per request basis.
+        /// The functions gets the HttpContext, and should return a boolean.
+        /// If functions returns true, the request is collected.
+        /// If functions returns false, the request is filtered out.
+        /// If filter throws exception, then this is considered as no filter being configured.
+        /// and requested is collected.
```

Comment:
This line doesn't seem to be right?

Reply:
User configured a filter.
Filter throws ex.
We cannot let the original request fails, so we have cover up the user's mistake. So we catch it and log it.

The next question is: What about the fallback behavior when users filter threw? We cannot know their original intention anyway. We can either
1. Move on as if filter functions was not configured at all. (This is the current PR)
2. Assume that user wanted to filter out and dont collect the behavior.

Are you suggesting option 2? Or something other ..

Reply:
This makes me think about the discussion in the last SIG meeting regarding security concerns and multiple ActivityListeners. That is, security may be one use case for filtering. Might be a reason to consider option 2.

Reply:
1. "If functions returns" -> "If filter returns".
2. "being configured. and requested is.." -> the `.` doesn't seem to be correct.
3. Probably could simplify this to "If filter returns false or throws exception, no telemetry data will be collected"

Reply:
Re worded.